### ISSUE TITLE
Feature: Grouping Permission Specification Sets into Business Roles in Identity Management Solution

### ISSUE THREAD

**Issue by brian-burton** [tmforum-oda/oda-canvas#82]
## Description

This issue is to add a BDD feature (in [compliance-test-kit/BDD-and-TDD/features](https://github.com/tmforum-oda/oda-canvas/tree/master/compliance-test-kit/BDD-and-TDD/features)) for **Feature: Grouping Permission Specification Sets into Business Roles in Identity Management Solution**.

This BDD would cover:

- Existing Permission Specification Sets in the component
- Creation/update/removal of corresponding technical roles in the identity platform
- Creation/update/removal of PartyRoles as business/job roles in the identity platform
- Creation/update/removal of mapping of technical roles to business roles in the identity platform
- Corresponding tests of authorisation in the component.

This links to [decision 2](https://github.com/tmforum-oda/oda-ca-docs/blob/master/Decision-Log/0002-role-vs-permission.md).

**Comment by pecanpete** [comment]
Do technical roles get associated with components .IF so the discussion in the Use role and permissions API  is to add in resource role as generalized term for thing that aren't parties which aligns with SID

**Comment by LesterThomas** [comment]
> Do technical roles get associated with components .IF so the discussion in the Use role and permissions API is to add in resource role as generalized term for thing that aren't parties which aligns with SID

Yes - the components expose technical roles towards the Canvas (In Role & Permission API these are described as `PermissionSpecSet`. The Identity Management service will be updated with these (automatically by the Identity operator) and the team operating the canvas platform can group technical roles into business roles and assign to users.
